cypress/integration/interface_builder_test.js

```diff
@@ -0,0 +1,35 @@
+describe('Interface builder tests', () => {
+  context('no access before login', () => {
+    it('redirects to home', () => {
+      cy.visit('/interfaces/new');
+      cy.location('pathname').should('eq', '/login');
+
+      cy.visit('/interfaces/testInterface');
+      cy.location('pathname').should('eq', '/login');
+    });
+  });
+
+  context('authenticated', () => {
+    beforeEach(() => {
+      cy.login();
+      cy.fixture('test.astarte.FirstInterface').as('test_interface');
+
+      cy.server();
+      cy.route('GET', '/realmmanagement/v1/*/interfaces/*/*', '@test_interface');
+    });
+
+    it('loads empty builder', () => {
+      cy.visit('/interfaces/new');
+      cy.location('pathname').should('eq', '/interfaces/new');
+      cy.get('#interfaceName').should('have.value', '');
+    });
+
+    it('shows selected interface', function() {
+      const testInterfaceName = this.test_interface.data.interface_name
+      const testInterfaceMajor = this.test_interface.data.version_major
+      cy.visit(`/interfaces/${testInterfaceName}/${testInterfaceMajor}`);
+      cy.location('pathname').should('eq', `/interfaces/${testInterfaceName}/${testInterfaceMajor}`);
+      cy.get('#interfaceName').should('have.value', testInterfaceName);
```

Comment:
can we also test major, minor and other properties?

Reply:
sure we can, i'll add more checks